### PR TITLE
Merge the Material and Cupertino progress bars

### PR DESCRIPTION
**Pull request by miDeb**
Still work in progress, so far I only merged the progress bars: The existing API still exists, but there is now a `VideoProgressBar` widget under the hood that takes a `ChewieProgressBarConfiguration` (for now).

### CODE REVIEW

**Comment by miDeb** [comment]
After thinking about this a bit more I have removed `ChewieProgressBarConfiguration`. I felt like introducing an additional object here is useless and confusing, so I made `barHeight`, `handleHeight` and `drawShadow` (these were previously the only members of `ChewieProgressBarConfiguration`) simply be constructor parameters.

**Comment by nstrelow** [comment]
Very amazing, this gets rid of so much code already 🤩 

I will review it soon. But actually I think this would make a good fist step PR.

E.g. first move progressbar and then the others.

**Comment by miDeb** [comment]
@nstrelow that would make sense as well. I made this ready for review.

**Comment by miDeb** [comment]
@nstrelow that would make sense as well. I made this ready for review.

File: lib/src/progress_bar.dart [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+class VideoProgressBar extends StatefulWidget {
+  VideoProgressBar(
+    this.controller, {
+    ChewieProgressColors? colors,
+    this.onDragEnd,
+    this.onDragStart,
+    this.onDragUpdate,
+    Key? key,
+    required this.barHeight,
+    required this.handleHeight,
```
Comment (miDeb) [inline lib/src/progress_bar.dart:15]: I only realized afterwards that `handleHeight` is `6` for both Material and Cupertino, but I then decided to leave it in.

Comment (nstrelow) [reply lib/src/progress_bar.dart:15]: That's fine!

**Comment by nstrelow** [comment]
In german I would say: Wunschlos Glücklich!

**Comment by miDeb** [comment]
Danke!

**Pull request merged**